UM-USElab/gradecraft-development
# Upload inline images in Froala

### Status
**READY**

### Description
* It should be possible for instructors and students to upload an inline image in the Froala editor. Currently uploading images does not work and is disabled. Now, uploading images is allowed and is uploaded to a new directory in the uploads folder (files/uploads/froala_uploads) and can be inserted as inline images in the editor. Text comments within submissions by students are  exported as pdfs and have inline image uploads attached and also preserve inline tables and formatting in the text comment.

### Related PRs
branch | PR
------ | ------
image-button | #4235 

### Gem dependencies
- ADDED [wkhtmltopdf-binary](https://github.com/zakird/wkhtmltopdf_binary_gem)
- ADDED [wicked_pdf](https://github.com/mileszs/wicked_pdf)

### Migrations
NO

### Steps to Test or Reproduce
* Uploading inline images
1. As an instructor or student access a page that has the Froala editor and allows images to be uploaded. Click the image button in the editor and upload an image. 
2. The image that is uploaded should be inserted as an inline image in the editor and it should be possible to view and edit the uploaded image even after navigating away and then back to the page where the image was uploaded in.

* Exporting text comments within submissions by students
1. As a student, submit an assignment with a text comments that has an inline image upload, an inline table and other formatting (any combination of these).
2. As an instructor, download all submissions for the assignment that the submission was made in. The downloaded archive should have text submissions exported as pdfs with formatting from the text comment as well as any inline images. Any comments that contain only text should also be exported as pdfs in the archive.

### Impacted Areas in Application
* Uploading inline images in the Froala editor
* Exporting text comments in submissions by students
* services/creates_submissions_export/create_submission_files.rb 

======================
Closes #4210
